hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/ArrayJdbcType.java

```diff
@@ -203,4 +203,24 @@ public String getFriendlyName() {
 	public String toString() {
 		return "ArrayTypeDescriptor";
 	}
+
+	/**
+	 * Check equality. Needed so that ArrayJdbcType in collections correctly match each other.
+	 *
+	 * @param o other object
+	 * @return true if the two array types share the same element type
+	 */
+	@Override
+	public boolean equals(Object o) {
+		if(o instanceof ArrayJdbcType) {
+			ArrayJdbcType other = (ArrayJdbcType) o;
+			return this.elementJdbcType.getJdbcTypeCode() == other.getElementJdbcType().getJdbcTypeCode();
+		}
+		return false;
```

Comment:
```suggestion
		if(o instanceof ArrayJdbcType) {
			ArrayJdbcType other = (ArrayJdbcType) o;
			return this.elementJdbcType.getJdbcTypeCode() == other.getElementJdbcType().getJdbcTypeCode();
		}
		return getClass() == o.getClass() && getElementJdbcType().equals( ((ArrayJdbcType) other).getElementJdbcType() );
```

Reply:
I feel like on reflection I should change this to be just your new line. As is if two different `JdbcType`s have the same type code my equals would incorrectly consider them equal.